src/redfishpower/redfishpower.c

```diff
@@ -314,6 +327,59 @@ static void powermsg_init_curl(struct powermsg *pm)
         Curl_easy_setopt((pm->eh, CURLOPT_HTTPGET, 1));
 }
 
+static char *resolve_hosts_url(const char *hostname, const char *path)
+{
+    char *url;
+    char ipstr[INET6_ADDRSTRLEN] = {0};
+    size_t len;
+    struct addrinfo *ai;
+    struct addrinfo *res = NULL;
+    int ret;
+
+    if ((ret = getaddrinfo(hostname, NULL, NULL, &res)))
+        err_exit(false, "getaddrinfo: %s", gai_strerror (ret));
+
+    for (ai = res; ai != NULL; ai = ai->ai_next) {
+        if (ai->ai_family == AF_INET) {
+            struct sockaddr_in addr;
+
+            memcpy(&addr, ai->ai_addr, ai->ai_addrlen);
+
+            if (!inet_ntop (AF_INET,
+                            &addr.sin_addr,
+                            ipstr,
+                            INET6_ADDRSTRLEN))
+                err_exit(true, "inet_ntop");
+
+            len = strlen("https://") + strlen(ipstr) + strlen(path) + 2;
+            url = xmalloc(len);
+            sprintf(url, "https://%s/%s", ipstr, path);
+            return url;
+        }
+        else if (ai->ai_family == AF_INET6) {
+            struct sockaddr_in6 addr6;
+
+            memcpy(&addr6, ai->ai_addr, ai->ai_addrlen);
+
+            if (!inet_ntop (AF_INET6,
+                            &addr6.sin6_addr,
+                            ipstr,
+                            INET6_ADDRSTRLEN))
+                err_exit(true, "inet_ntop");
+
+            len = strlen("https://") + strlen(ipstr) + strlen(path) + 2;
+            url = xmalloc(len);
+            sprintf(url, "https://%s/%s", ipstr, path);
```

Comment:
Any reason not to use `asprintf()` here and the other places?

Reply:
Heh, for some reason I added an asprintf equivalent function in libcommon called `hsprintf()`.   it calls xmalloc internally so you don't have to check for errors I guess.

Seems like long term that should go away and be replaced wtih asprintf though.

Reply:
hmmmm, i can't remember why i didn't use it, probably b/c i wanted to `xmalloc()` and didn't know there was a `xasprintf()` like equivalent.

are you saying use `asprintf()` directly?  Or perhaps we make a `xasprintf()`?

Reply:
It's pretty easy to check the return code from asprintf and exit if it fails so I'd vote for that.

Reply:
ahhh yeah, we should replace with a `xasprintf()` equivalent.  lets make that a cleanup todo.  I'll write up an issue.

Reply:
ahhh page didn't refresh and didn't see your comment.  I think i'll leave that cleanup till another day when `hsprintf()` can be removed / cleaned up as well.